Makefile

```diff
@@ -52,25 +55,26 @@ bootstrap: toolset
 test:
 	build/scripts/tests.sh $(BUILD_CONFIGURATION)
 
+restore: $(RESTORE_SEMAPHORE_PATH)
+
+$(RESTORE_SEMAPHORE_PATH): $(ROSLYN_TOOLSET_PATH)
+	@build/scripts/restore.sh $(ROSLYN_TOOLSET_PATH) && \
+	touch $(RESTORE_SEMAPHORE_PATH)
```

Comment:
Looks like this means `make restore` will only restore once.  If I change a `project.json` then `make restore` won't do anything.  I think that's fine as a first step but we should file a bug to fix this in the future.